### PR TITLE
Slight refactor of logging samples.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: logging/logs.js [hunk 1]
```diff
@@ -13,114 +13,136 @@
...
- * @param {string} options.name The name of the log to write to.
- * @param {object} options.resource The resource to associate with the log entry.
- * @param {string|object} options.entry The body of the log entry.
- * @param {function} callback The callback function.
- */
-function writeLogEntry (options, callback) {
-  // Get a reference to an existing log
-  var log = logging.log(options.name);
+
+function writeLogEntryAdvanced (logName, options, callback) {
```
Comment (ace-n) [inline logging/logs.js]: Should we have `logName` **and** `options` here?

File: logging/logs.js [hunk 2]
```diff
@@ -13,114 +13,136 @@
...
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Found %d entries!', entries.length);
+    return callback(null, entries);
+  });
+}
+
+function listLogEntriesAdvanced (filter, limit, sort, callback) {
```
Comment (ace-n) [inline logging/logs.js]: Is there any benefit to making these parameter names the same as their respective `options` values?

File: logging/logs.js (hunk 1 above)
Comment (jmdobry) [inline logging/logs.js]: Required things go into positional arguments, optional stuff into an `options` argument.

File: logging/logs.js (hunk 2 above)
Comment (jmdobry) [inline logging/logs.js]: Sure why not

File: logging/system-test/logs.test.js [hunk 3]
```diff
@@ -22,48 +22,51 @@ var filter = 'resource.type="global" AND logName="projects/' + projectId + '/log
...
-        assert.ifError(err);
-      }
-      done();
+  describe('deleteLog', function () {
+    it('should delete a log', function (done) {
+      program.deleteLog(logName, function (err, apiResponse) {
+        // Ignore "Not Found" error
+        if (err && err.code !== 404) {
+          assert.ifError(err);
+          assert(apiResponse);
```
Comment (ace-n) [inline logging/system-test/logs.test.js]: Remove this 'naked assert'? e.g.
```javascript
assert.notEqual(apiResponse, null)
```

File: logging/system-test/sinks.test.js [hunk 4]
```diff
@@ -116,8 +107,9 @@ describe('logging:sinks', function () {
 
   describe('deleteSink', function () {
     it('should delete a sink', function (done) {
-      program.deleteSink(sinkName, function (err) {
+      program.deleteSink(sinkName, function (err, apiResponse) {
         assert.ifError(err);
+        assert(apiResponse);
```
Comment (ace-n) [inline logging/system-test/sinks.test.js]: Naked assert?

File: logging/test/logs.test.js [hunk 5]
```diff
@@ -16,14 +16,21 @@
...
 
 function getSample () {
+  var apiResponseMock = {};
   var entriesMock = [{}];
   var entryMock = {};
   var logMock = {
     entry: sinon.stub().returns(entryMock),
-    write: sinon.stub().callsArgWith(1, null),
-    delete: sinon.stub().callsArgWith(0, null)
+    write: sinon.stub().callsArgWith(1, null, apiResponseMock),
```
Comment (ace-n) [inline logging/test/logs.test.js]: I assume this is what you wanted to fix in a separate mini-PR. (That's fine by me.)

**Comment by ace-n** [comment]
LGTM, assuming the naked asserts aren't an issue (or will be resolved in a separate PR).

File: logging/test/logs.test.js (hunk 5 above)
Comment (jmdobry) [inline logging/test/logs.test.js]: Yeah.

**Pull request merged**